### PR TITLE
Add tests for CallbackThrottleCreator

### PR DESCRIPTION
**Pull request by Mecharyry**
Just adding tests for the `CallbackThrottleCreator` 😄

### CODE REVIEW

File: library/src/main/java/com/novoda/downloadmanager/CallbackThrottleCreator.java [hunk 1]
```diff
@@ -47,7 +47,7 @@ CallbackThrottle create() {
             case CUSTOM:
                 return createCallbackThrottle();
             default:
-                throw new IllegalStateException("type " + type + " not supported");
+                throw new IllegalStateException("type " + type + " not supported.");
```
Comment (jonreeve) [inline library/src/main/java/com/novoda/downloadmanager/CallbackThrottleCreator.java]: Grammar! 😄 👍

File: library/src/test/java/com/novoda/downloadmanager/CallbackThrottleCreatorTest.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import java.util.concurrent.TimeUnit;
+
+import org.junit.Test;
+
+import static com.google.common.truth.Truth.assertThat;
+
+public class CallbackThrottleCreatorTest {
+
+    @Test
+    public void createsTimeThrottle() {
```
Comment (jonreeve) [inline library/src/test/java/com/novoda/downloadmanager/CallbackThrottleCreatorTest.java]: ✨

File: library/src/test/java/com/novoda/downloadmanager/CallbackThrottleCreatorTest.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void throwsException_whenCustomCallbackIsNotFound() {
+        CallbackThrottleCreator.ByCustomThrottle(TestNotFoundCustomThrottle.class)
+                .create();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void throwsException_whenCustomCallbackIsNotInstantiable() {
```
Comment (jonreeve) [inline library/src/test/java/com/novoda/downloadmanager/CallbackThrottleCreatorTest.java]: Are these things we have code to specifically handle? They sound a bit like language features, as in not our code. If we have cases specifically for them then OK.

Comment (Mecharyry) [reply library/src/test/java/com/novoda/downloadmanager/CallbackThrottleCreatorTest.java]: We have cases for these in the code. I still don't fully understand the need to pass a class type and instantiate in the `CallbackThrottleCreator` but it's there with all the associated catches 😄

Comment (jonreeve) [reply library/src/test/java/com/novoda/downloadmanager/CallbackThrottleCreatorTest.java]: yeah it does seem odd but :man_shrugging: 

Cool, so long as it's testing our stuff 👍

File: library/src/test/java/com/novoda/downloadmanager/CallbackThrottleCreatorTest.java [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+    @Test(expected = RuntimeException.class)
+    public void throwsException_whenCallbackThrottleDoesNotExist() {
+        CallbackThrottleCreator.ByCustomThrottle(null)
+                .create();
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void throwsException_whenCustomCallbackIsNotPublic() {
+        CallbackThrottleCreator.ByCustomThrottle(TestNonPublicCustomThrottle.class)
```
Comment (zegnus) [inline library/src/test/java/com/novoda/downloadmanager/CallbackThrottleCreatorTest.java]: 💯

**Pull request merged**